### ISSUE TITLE
Make the create user scripts idempotent.

### ISSUE THREAD

**Issue by stvoutsin** [wfau/aglais#792]
In the latest deploy, when attempting to run the "createusermain" script which we already create linux accounts (zrq, stv, nch, dcr), we get an error:
```
adduser: user 'zrq' already exists
mkdir: cannot create directory ‘/home/zrq/.ssh’: File exists
```
The shiro user is create, and the user can login, but when attempting to run a notebook we get a permission denied error, when the user impersonated job tries to access the conf/zeppelin-env.sh file. This implies that the either the user group or ssh access was not setup correctly.

**Fix:**
Make the create user scripts idempotent. 
Perhaps remove linux user creation from deployment script, this should probably be done from the createusermain script only.
Modify createusermain to check for existing (linux?) user

**Comment by Zarquan** [comment]
Modify all of the individual scripts to check if the thing they are creating already exists.
* `create_hdfs_home`
* `create_mysql_user`
* `create_unix_user`
* `create_notebook_clone`

**Comment by Zarquan** [comment]
Split into separate tasks for each to make it easier to develop in stages.

**Comment by Zarquan** [comment]
Fixed in #846

**Comment by Zarquan** [comment]
Fixed by #846
